File: ecos-records/src/main/java/ru/citeck/ecos/records3/rest/v1/delete/DeleteResp.java
package ru.citeck.ecos.records3.rest.v1.delete;

import lombok.Getter;
import lombok.Setter;
import ru.citeck.ecos.records3.record.op.delete.dto.DelStatus;
import ru.citeck.ecos.records3.rest.v1.RequestResp;

import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
public class DeleteResp extends RequestResp {

    private List<DelStatus> statuses = new ArrayList<>();

    public void setStatuses(List<DelStatus> statuses) {
        if (statuses != null) {
            this.statuses = new ArrayList<>(statuses);
        } else {
            this.statuses.clear();
        }
        this.statuses = statuses;
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records3/rest/v1/delete/DeleteResp.kt
package ru.citeck.ecos.records3.rest.v1.delete

import ru.citeck.ecos.records3.record.dao.delete.DelStatus
import ru.citeck.ecos.records3.rest.v1.RequestResp
import java.util.*

class DeleteResp : RequestResp() {

    var statuses: MutableList<DelStatus> = ArrayList<DelStatus>()
        private set

    fun setStatuses(statuses: List<DelStatus>?) {
        if (statuses != null) {
            this.statuses = ArrayList(statuses)
        } else {
            this.statuses = ArrayList()
        }
    }
}
